astroplan/core.py

```diff
@@ -25,82 +40,137 @@ class Observer(object):
     Some comments.
     """
 
-    def __init__(self, name, longitude, latitude, elevation, pressure,
-                 relative_humidity, temperature, timezone, description=None):
+    def __init__(self, name=None, location=None, latitude=None, longitude=None,
+                 elevation=None, timezone='UTC', pressure=None,
+                 relative_humidity=None, temperature=None, description=None):
         """
         Initializes an Observer object.
 
-        <longer description>
+        TODO: <longer description>
 
         Parameters
         ----------
         name : str
             A short name for the telescope, observatory or location.
 
-        longitude : str or `~astropy.units.quantity`?
-            The longitude of the observing location.
+        pressure : `~astropy.units.Quantity`
+            The ambient pressure.
 
-        latitude : str or `~astropy.units.quantity`?
-            The latitude of the observing location.
+        location : `~astropy.coordinates.EarthLocation`
+            The location (latitude, longitude, elevation) of the observatory.
+
+        longitude : str or `~astropy.units.Quantity`
+            The longitude of the observing location. If str, should be a string
+            that initializes a `~astropy.coordinates.Longitude` object with
+            units in degrees.
+
+        latitude : str or `~astropy.units.Quantity`
+            The latitude of the observing location. If str, should be a string
+            that initializes a `~astropy.coordinates.Latitude` object with
+            units in degrees.
 
         elevation : `~astropy.units.Quantity`
             The elevation of the observing location, with respect to sea
             level.
 
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
-
         relative_humidity : float
             The ambient relative humidity.
 
         temperature : `~astropy.units.Quantity`
             The ambient temperature.
 
-        timezone : WHAT TYPE IS pytz.timezone ?
-            The local timezone.
+        timezone : str or `datetime.tzinfo`
+            The local timezone, as either an instance of `pytz.timezone()` or
+            the string accepted by `pytz.timezone()`.
 
         description : str
             A short description of the telescope, observatory or observing
             location.
         """
-        raise NotImplementedError()
 
-    def set_environment(self, pressure, relative_humidity, temperature):
-        """
-        Updates the Observer object with environmental conditions.
+        if name is not None:
+            self.name = name
 
-        <longer description>
+        if pressure is None:
+            pressure = 0
 
-        Parameters
-        ----------
-        pressure : `~astropy.units.Quantity`
-            The ambient pressure.
+        if temperature is None:
+            temperature = 0 * u.deg_C
```

Comment:
Deal. Setting *pressure* to zero by default is pretty common to specify an atmosphere-less calculation – we're still OK on that though, right?